python/Dockerfile

```
@@ -70,7 +70,7 @@ RUN PYENV_VERSION=$PY_3_8 pyenv exec python --version | grep "Python $PY_3_8" ||
 RUN bash /opt/python/helpers/build $PY_3_8
 # This python environment occupies ~0.5 GB and gets used for a fraction of jobs, so store it compressed.
 RUN cd $PYENV_ROOT/versions \
-  && tar czf $PY_3_8.tar.gz $PY_3_8
+  && tar --mtime=./$PY_3_8/lib/libpython3.so --sort=name -czf $PY_3_8.tar.gz $PY_3_8
```

```suggestion
  # using a consistent mtime and sort order increases the odds this layer will be pulled from the docker image cache
  && tar --mtime=./$PY_3_8/lib/libpython3.so --sort=name -czf $PY_3_8.tar.gz $PY_3_8
```

Can you add a code comment like ☝️ to all of these for consistency? I know there's `git blame`, but this one is non-obvious enough that an inline comment would be helpful for anyone who comes along later wondering why this was done and if changing it will break anything.

---

I can, and I will!